src/JSInterop/Microsoft.JSInterop/src/Infrastructure/ByteArrayJsonConverter.cs

```diff
@@ -76,7 +76,7 @@ public ByteArrayJsonConverter(JSRuntime jSRuntime)
 
     public override void Write(Utf8JsonWriter writer, byte[] value, JsonSerializerOptions options)
     {
-        var id = ++_byteArrayId;
+        var id = Interlocked.Increment(ref _byteArrayId);
```

Comment:
TODO: need to confirm using `[System.Threading](https://docs.microsoft.com/en-us/dotnet/api/system.threading?view=net-6.0)` doesn't cause issues for `wasm`.